### PR TITLE
Add live update and output sync support

### PR DESCRIPTION
**Pull request by nickmertin**
Live update is provided by the `NewChanges` message. Output sync is provided by the `ScriptOutput` message.

### CODE REVIEW

File: src/proto/script_host.proto [hunk 1]
```diff
@@ -122,7 +125,14 @@ message TwoFactorResponse {
     string two_factor = 2;
 }
 
-// Next: 8
+// Next: 2
+// Client message to update the local changes to the repository
+message NewChanges {
+    // A Git patch file containing the update
+    bytes patch = 1;
+}
```
Comment (Octogonapus) [inline src/proto/script_host.proto]: This seems like the same thing as a patch in the GitFS sense. Is that right? If so, could this script proto and the gitfs proto share a patch message?

Comment (nickmertin) [reply src/proto/script_host.proto]: Yeah, we can do that. I'd still like to keep a `NewChanges` message here, as that makes it easier to change the structure of this command in the future.

Comment (Octogonapus) [reply src/proto/script_host.proto]: Without `NewChanges`, it wouldn't be any harder to change than `ProjectSpec`, right? It feels inconsistent to me. Should we add another level of indirection to `ProjectSpec`?

Comment (nickmertin) [reply src/proto/script_host.proto]: Look at ProjectSpec - I did.

Comment (Octogonapus) [reply src/proto/script_host.proto]: The levels in ProjectSpec are ProjectSpec -> Patch -> bytes
The levels in SessionClientMessage are SessionClientMessage -> NewChanges -> Patch -> bytes

My point is: can we have SessionClientMessage -> Patch -> bytes instead?

Comment (nickmertin) [reply src/proto/script_host.proto]: Ah. Yes, I suppose that works. It just means that if we ever want to add something to that command, we run into compatibility problems.

File: src/proto/robot_config.proto [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package bowler.robot_config;
+
+import "git_f_s.proto";
+
```
Comment (Octogonapus) [inline src/proto/robot_config.proto]:
```suggestion
```

**Review by Octogonapus — APPROVED** [review]

**Review by Octogonapus — CHANGES_REQUESTED** [review]
We also need to support sending a new robot config from the server to the client.

**Review by Octogonapus — APPROVED** [review]

**Comment by Octogonapus** [comment]
Is this RTM?

**Pull request merged**